clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -390,15 +390,9 @@ void executeCommandWithRetry(
       Runnable mainProcessCommandRunnable) {
     boolean isTopicPartitionHosted = isTopicPartitionHosted(topicName, partition);
 
-    // drop storage partition in main process if it does not exist
-    if (command == REMOVE_PARTITION && !isTopicPartitionHosted) {
-      mainProcessCommandRunnable.run();
-      return;
-    }
-
     do {
       if (isTopicPartitionHostedInMainProcess(topicName, partition)
-          || !isTopicPartitionHosted && command != START_CONSUMPTION) {
+          || (!isTopicPartitionHosted && command != START_CONSUMPTION && command != REMOVE_PARTITION)) {
```

Comment:
Can you add some javadoc here? I am not sure I fully understand this logic.

Reply:
Updated the Java Doc. Please feel free to take a 2nd look when you get some time.

Reply:
I understand for `START_PARTITION`, it is possible since Venice Server would like to bring up a new partition replica, but how can it be possible for `REMOVE_PARTITION`?

Reply:
The Helix might want to drop a stale topic partition's replica, so it could send an OFFLINE->DROPPED message. The message could arrive when server is restarted. Since we changed the behavior to restore data partition in II, the message should arrive at II process.

Reply:
isTopicPartitionHosted means if the TP has ever gone through the subscription flow (OFFLINE->STANDBY)
Since for the case, it is only the data RocksDB opened by II process, not going through the flow, and no metadata has been updated, so isTopicPartitionHosted will be false.

Reply:
I see.
Please add the above statement as javadoc and it is not clear from the naming of this variable.

Reply:
Yeah, updated the Java doc as well.